components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -265,6 +265,46 @@ public Response addCommentToAPI(String apiId, PostRequestBodyDTO postRequestBody
         return null;
     }
 
+    /**
+     * Retrieves the async api specification document of an API
+     *
+     * @param apiId API identifier
+     * @param environmentName name of the gateway environment
+     * @param ifNoneMatch If-None-Match header value
+     * @param xWSO2Tenant requested tenant domain for cross tenant invocations
+     * @param messageContext CXF message context
+     * @return Async API Specification document of the API for the given cluster or gateway environment
+     */
+    @Override
+    public Response apisApiIdAsyncApiSpecificationGet(String apiId, String environmentName, String ifNoneMatch, String xWSO2Tenant, MessageContext messageContext) throws APIManagementException {
+            try {
+                String organization = RestApiUtil.getValidatedOrganization(messageContext);
+                APIConsumer apiConsumer = RestApiCommonUtil.getLoggedInUserConsumer();
+
+                API api = apiConsumer.getLightweightAPIByUUID(apiId, organization);
+                if (api.getUuid() == null) {
+                    api.setUuid(apiId);
+                }
+
```

Comment:
Shall we remove unnecessary empty line?

Reply:
This has been already changed from another PR